lib/ansible/module_utils/basic.py

```diff
@@ -167,6 +167,8 @@
 from ansible.module_utils.six.moves import map, reduce
 from ansible.module_utils._text import to_native, to_bytes, to_text
 
+PASSWORD_MATCH = re.compile(r'^(?:.+[-_\s])?pass(?:[-_\s]?(?:word|phrase|wrd|wd)?)(?:[-_\s].+)?$', re.I)
```

Comment:
Keep the two PASSSWORD regexes separate since they're operating on two different sets of data with separate constraints.  It's hard enough trying to understand regexes without the regex having conditions that are applicable to a different problem then the code you're looking at.

Do keep them close together and document a bit about the cases that they're trying to solve, though.

Differences between the two: The match for module option names cannot use "-" and space.

The match for command line args should not be anchored.  We're really just scanning for "password", "passphrase", etc).  The current regex misses things that could get passed in on the command line like "--password=foo" and "--input=<java><password>Why?Why?Why?</password></java>"